purchase_request/tests/test_purchase_request_procurement.py

```diff
@@ -110,3 +113,39 @@ def test_procure_purchase_request(self):
             "Test Test, Split", "Test, Split", self.product_1, 10
         )
         self.procurement_group_run("Test Test, Split", False, self.product_1, 10)
+
+    @freeze_time("2022-03-15 01:00:00")
+    def test_duplicated_purchase_request(self):
+        self.assertEqual(self.env["purchase.request"].search_count([]),0)
+        has_route = self.procurement_group_run(
+            "Test Purchase Request Single Line",
+            "Test Purchase Request Single Line",
+            self.product_1,
+            4,
+        )
+        self.assertTrue(has_route)
+        self.env["procurement.group"].run_scheduler()
+        pr = self.env["purchase.request"].search(
+            [("origin", "=", "Test Purchase Request Single Line")]
+        )
+        self.assertTrue(pr.to_approve_allowed)
+        self.assertEqual(pr.origin, "Test Purchase Request Single Line")
+
+        prl = self.env["purchase.request.line"].search([("request_id", "=", pr.id)])
+        self.assertEqual(prl.request_id, pr)
+        self.assertEqual(prl.product_qty, 4)
+        self.assertEqual(prl.product_id, self.product_1)
+
+        # another request
+        has_route = self.procurement_group_run(
+            "Test Purchase Request Single Line",
+            "Test Purchase Request Single Line",
+            self.product_1,
+            5,
+        )
+        self.assertTrue(has_route)
+        self.env["procurement.group"].run_scheduler()
+        # make sure no new PR is created
+        self.assertEqual(self.env["purchase.request"].search_count([]),1)
+        self.assertEqual(len(pr.line_ids), 1)
```

Comment:
@simahawk @gurneyalex  any idea why it creates 2nd line here even if it should not?  Getting `AssertionError: 2 != 1`. Not sure if whole TC scenario is correct but looking at the model code, there should be procurement handling in the background - didn't find better starting point.

Reply:
wild guess: the procurement group date is not frozen?

Reply:
@simahawk in filtered domain i was comparing datetime.datetime with datetime.date, just fixed by extracting date(), let's see.

```
-> if same_product_date_request_line:
(Pdb) pr.line_ids
purchase.request.line(72,)
(Pdb) pr.line_ids.date_required
datetime.date(2022, 3, 16)
(Pdb) request_line_data["date_required"]
datetime.datetime(2022, 3, 16, 8, 15, 14)
```